syntax = "proto3";

package users;

service UserService {
    rpc CreateUser(CreateUserRequest) returns (CreateUserResponse);
    rpc GetAllUsers(GetAllUsersRequest) returns (GetAllUsersResponse);
    rpc GetUser(GetUserRequest) returns (GetUserResponse);
    rpc UpdateUser(UpdateUserRequest) returns (UpdateUserResponse);
    rpc DeleteUser(DeleteUserRequest) returns (DeleteUserResponse);
}

message CreateUserRequest {
    string name = 1;
    string email = 2;
}

message CreateUserResponse {
    string message = 1;
}

message GetAllUsersRequest {}

message User {
    int32 id = 1;
    string name = 2;
    string email = 3;
}

message GetAllUsersResponse {
    repeated User users = 1;
}

message GetUserRequest {
    int32 user_id = 1;
}

message GetUserResponse {
    User user = 1;
    string message = 2;
}

message UpdateUserRequest {
    int32 user_id = 1;
    string name = 2;
    string email = 3;
}

message UpdateUserResponse {
    string message = 1;
}

message DeleteUserRequest {
    int32 user_id = 1;
}

message DeleteUserResponse {
    string message = 1;
}


message Empty {}
